web/packages/design/src/CollapsibleInfoSection/CollapsibleInfoSection.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  PropsWithChildren,
+  useId,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import styled from 'styled-components';
+
+import { Box, Button, Text } from 'design';
+import { BoxProps } from 'design/Box';
+import { Minus, Plus } from 'design/Icon';
+
+type CollapsibleInfoSectionProps = {
+  /* defaultOpen is optional and determines whether the section is open or closed initially */
+  defaultOpen?: boolean;
+  /* onClick is an optional callback for when the toggle is clicked */
+  onClick?: (isOpen: boolean) => void;
```

Comment:
In what scenario would we not want to provide `onClick` props for component that expands/collapses on click?

Reply:
Not sure I understand? The open/collapse logic is handled internally, but I figured there may be some use-cases where it'd be helpful for the consumer to be able to provide a callback for when the section is opened/closed

Reply:
Ah got it. So my comment is invalid cause I thought this is the main onClick handler.